### ISSUE TITLE
`yarn rwfw project:sync` rebuilds N times if N files change

### ISSUE THREAD

**Issue by nickpdemarco** [redwoodjs/graphql#4965]
If you simultaneously change more than one file (say, with `git stash` or `git stash pop`), the syncing engine will enqueue N rebuild tasks unnecessarily. This is a low-impact bug in my opinion, but does slow down development when A/B testing a stashed set of changes.

# Reproducing
```bash
$ yarn create redwood-app x-test-app
$ cd x-test-app
$ RWFW_PATH=/path/to/rw yarn rwfw project:sync
```
Then, in the RW repo:
```bash
# Change more than one file
$ git stash
# Watch multiple rebuilds in the x-test-app terminal
```

**Comment by jtoar** [comment]
@nickpdemarco definitely, we should either debounce or cancel previous rebuilds if others are queued.

**Comment by Josh-Walker-GM** [comment]
Debounce would typically prevent newer operations in favour of older ones where as here we wish to prioritise newer events, yes?

I assume [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) can't be used here given its node 15 requirement?
